### PR TITLE
Source specific and error validation rebuild tests for CASSANDRA-9875

### PR DESCRIPTION
**Pull request by geoffxy**
These are new dtests to go along with the changes in [CASSANDRA-9875](https://issues.apache.org/jira/browse/CASSANDRA-9875).

### CODE REVIEW

File: rebuild_test.py [hunk 1]
```diff
@@ -290,3 +290,217 @@ def rebuild_ranges_test(self):
...
+        self.create_ks(session, 'ks1', {'dc1': 1, 'dc2': 1})
+        self.create_cf(session, 'cf', columns={'c1': 'text', 'c2': 'text'})
+        insert_c1c2(session, n=keys, consistency=ConsistencyLevel.ALL)
+        # ks2 will not be rebuilt in node3
+        self.create_ks(session, 'ks2', {'dc1': 1, 'dc2': 1})
+        self.create_cf(session, 'cf', columns={'c1': 'text', 'c2': 'text'})
+        insert_c1c2(session, n=keys, consistency=ConsistencyLevel.ALL)
+        session.shutdown()
+
+        # Bootstraping a new node in dc3 with auto_bootstrap: false
```
Comment (ptnapoleon) [inline rebuild_test.py]: nit `Bootstrapping`

**Comment by ptnapoleon** [comment]
Looks great! +1, with or without that typo nit

**Comment by mambocab** [comment]
By and large very good, yes! I'm working on a commit with a couple fixes and simplifications, so please hold off on merge.

**Comment by mambocab** [comment]
Here we go:

https://github.com/geoffxy/cassandra-dtest/compare/CASSANDRA-9875...mambocab:1273_use-populate

Makes a couple simplifications:

- Manually creating the nodes is unnecessary in at least some of these tests. It's done elsewhere in the suite to allow the test to stop and start nodes while populating, but yours don't seem to do that. These changes provide a template for the other tests to follow.
- There's an `assertRaises` variant that checks the error against a regex, and I've used it here.

Could you use these across all the new tests please?

In addition, the names of the tests you added don't contain the string `test` -- could you add `_test` to the end of each?

**Comment by mambocab** [comment]
Also, this change: https://github.com/geoffxy/cassandra-dtest/compare/CASSANDRA-9875...mambocab:1273_use-populate#diff-6aebd89fab59ec5fcb287f22e79554abR325 is just for readability -- it shouldn't change the test's behavior.

**Comment by ptnapoleon** [comment]
To add to mambocab's point, the string `test` should be in each test name so that nosetests properly detects and runs those methods as tests

**Comment by geoffxy** [comment]
Thanks for the review and sorry for the delay in making the changes! I've updated the test names to include a `_test` suffix, fixed the typo, and applied @mambocab's simplifications across the tests. Please let me know how it looks.

**Comment by mambocab** [comment]
Sorry for the delay in rereview, I was at Cassandra Summit. Code looks great. Running the new tests here: https://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/

**Comment by geoffxy** [comment]
No worries about the delay! I've rebased my branch to resolve the merge conflict. It looks like it was just a conflict in the imports.

**Comment by mambocab** [comment]
Sorry for the review delay. I'm running on CassCI here: http://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/332/

**Comment by mambocab** [comment]
I needed to rebase to get this running because of changes in `tools`, but that shouldn't be necessary for merging.

**Comment by mambocab** [comment]
Oh, and here's the new build: http://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/333/

**Comment by mambocab** [comment]
This needed another change to account for changes to the `Tester` API:

https://github.com/riptano/cassandra-dtest/commit/f27714dd4b71fcb4eb380f8c5ebaa78a69c2c3a5

You'll have to account for this before merge; sorry about that. Re-running CI here:

http://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/334/

**Comment by geoffxy** [comment]
No worries - I've rebased and added your commit.

**Comment by mambocab** [comment]
Passes on `trunk`: http://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/360/

I'm +1, any objections to merge?

**Comment by ptnapoleon** [comment]
Not from me, but i'm not sure if you were asking me or @geoffxy

**Comment by mambocab** [comment]
@geoffxy has suffered enough -- I'll merge.

Thanks for the tests, @geoffxy!

**Pull request merged**